### PR TITLE
feat: migrate to disnake and dpy 2.0

### PR DESCRIPTION
**Pull request by onerandomusername**
As discussed in dev-contrib: 
image

### CODE REVIEW

File: bot/bot.py [hunk 1]
```diff
@@ -88,7 +88,7 @@ async def startup_greeting(self) -> None:
         """Announce presence to the devlog channel."""
         embed = Embed(description="Connected!")
         embed.set_author(
-            name="Gurkbot", url=constants.BOT_REPO_URL, icon_url=self.user.avatar_url
+            name="Gurkbot", url=constants.BOT_REPO_URL, icon_url=self.user.avatar.url
```
Comment (Arnav-2004) [inline bot/bot.py]: Everything works fine, just check if the bot has profile picture or not and we will be good to go!

**Review by Arnav-2004 — APPROVED** [review]
LGTM!

File: bot/bot.py (hunk 1 above)
Comment (onerandomusername) [reply bot/bot.py]: https://github.com/gurkult/gurkbot/pull/155/commits/b1c35161ace78136d2e408c60a54b3b21dce7692 :D

Comment (gustavwilliam) [reply bot/bot.py]: Does this fix the "avatar not found" issue we talked about in #dev-gurkbot?

Comment (onerandomusername) [reply bot/bot.py]: https://github.com/gurkult/gurkbot/commit/b1c35161ace78136d2e408c60a54b3b21dce7692 fixes it, yes

**Review by Akarys42 — APPROVED** [review]
Let's try it

File: bot/exts/utils/bot_stats.py [hunk 2]
```diff
@@ -32,15 +32,15 @@ async def stats(self, ctx: commands.Context) -> None:
...
 
-        embed.set_thumbnail(url=self.bot.user.avatar_url)
+        embed.set_thumbnail(url=self.bot.user.display_avatar.url)
 
         uptime = humanize.precisedelta(
             datetime.utcnow().timestamp() - self.bot.launch_time
         )
 
         fields = {
             "Python Version": python_version(),
```
Comment (gustavwilliam) [inline bot/exts/utils/bot_stats.py:42]:
```suggestion
            "Python version": python_version(),
```
I think this looks nicer

File: bot/exts/utils/bot_stats.py [hunk 3]
```diff
@@ -32,15 +32,15 @@ async def stats(self, ctx: commands.Context) -> None:
...
-            "discord.py Version": __version__,
+            "disnake Version": __version__,
```
Comment (gustavwilliam) [inline bot/exts/utils/bot_stats.py:43]:
```suggestion
            "Disnake version": __version__,
```
Looks nicer imo.

**Review by gustavwilliam — COMMENTED** [review]
Nitpicking before testing in production :eyes:

File: bot/exts/utils/bot_stats.py (hunk 3 above)
Comment (onerandomusername) [reply bot/exts/utils/bot_stats.py]: https://github.com/gurkult/gurkbot/pull/155/commits/7ab5b706e62acc02ae59d3e658b242f9c8c728f9

File: bot/exts/utils/bot_stats.py (hunk 2 above)
Comment (onerandomusername) [reply bot/exts/utils/bot_stats.py:42]: https://github.com/gurkult/gurkbot/pull/155/commits/7ab5b706e62acc02ae59d3e658b242f9c8c728f9

Comment (onerandomusername) [reply bot/exts/utils/bot_stats.py:42]: https://github.com/gurkult/gurkbot/pull/155/commits/7ab5b706e62acc02ae59d3e658b242f9c8c728f9

**Comment by gustavwilliam** [comment]
@Akarys42 let's merge this tomorrow (technically today) then

**Pull request merged**